api_tests/e2e/test_creator.ts

```diff
@@ -0,0 +1,210 @@
+            default:
+                throw new Error("Unexpected error: unknown method");
+        }
+    })(getMethod(actionTrigger.action));
+}
+
+export async function createTests(
+    alice: Actor,
+    bob: Actor,
+    actions: ActionTrigger[],
+    initialUrl: string,
+    listUrl: string,
+    initialRequest: object
+) {
+    // This may need to become more generic at a later stage
+    // However, it would be unnecessary pre-optimisation now.
+    let swapLocations: { [key: string]: string } = {};
+
+    it(
+        "[alice] Should be able to make a request via HTTP api to " +
+            initialUrl,
+        async () => {
+            let res: ChaiHttp.Response = await chai
+                .request(alice.comit_node_url())
+                .post(initialUrl)
+                .send(initialRequest);
+            res.should.have.status(201);
+            const swapLocation: string = res.header.location;
+            swapLocation.should.not.be.empty;
+            swapLocations["alice"] = swapLocation;
+        }
+    );
+
+    it("[bob] Shows the Swap as IN_PROGRESS in " + listUrl, async () => {
+        let body = (await bob.pollComitNodeUntil(
+            listUrl,
+            body => body._embedded.swaps.length > 0
+        )) as SwapsResponse;
+
+        const swapEmbedded = body._embedded.swaps[0];
+        swapEmbedded.protocol.should.equal("rfc003");
+        swapEmbedded.status.should.equal("IN_PROGRESS");
+        const swapLink = swapEmbedded._links;
+        swapLink.should.be.a("object");
+        const swapLocation: string = swapLink.self.href;
+        swapLocation.should.not.be.empty;
+        swapLocations["bob"] = swapLocation;
+    });
+
+    while (actions.length !== 0) {
+        let action = actions.shift();
+        let actionHref: string = null;
+        let actionDirective: ActionDirective = null;
+        let actionExecutionResult: any = null;
+
+        it(
+            "[" +
+                action.actor.name +
+                "] Can get the " +
+                action.action +
+                " action",
+            async function() {
+                this.timeout(action.timeout || 10000);
+                [actionHref, actionDirective] = await getAction(
+                    action.actor,
+                    swapLocations[action.actor.name],
+                    action
+                );
+            }
+        );
+
+        it(
+            "[" +
+                action.actor.name +
+                "] Can execute the " +
+                action.action +
+                " action",
+            async function() {
+                actionExecutionResult = await executeAction(
+                    action.actor,
+                    action,
+                    actionHref,
+                    actionDirective
+                );
+            }
+        );
+
+        let body: any = null;
+        if (action.state) {
+            it(
+                "[" + action.actor.name + "] state is as expected",
```

Comment:
Nice. I actually didn't realise when I suggested it that a state assertion and an action execution could be rolled into the same object (with the state assertion happening after).

🔧 I'd prefer the text:
```
`[  ${action.actor.name}  ] transitions to correct state`
```

🤔It would be nice if we could say which state we are waiting for. We could do this by using something like: https://github.com/dchester/jsonpath#readme and then specifying the predicate as `"$.beta_ledger.status", "Redeemed"`.